frontend/src/app/lib/addon.js

```diff
@@ -7,15 +7,19 @@ import experimentActions from '../actions/experiments';
 import { getExperimentByID, getExperimentByURL, getExperimentInProgress } from '../reducers/experiments';
 
 const INSTALL_STATE_WATCH_PERIOD = 2000;
-
 const MOZADDONMANAGER_ALLOWED_HOSTNAMES = [
   'testpilot.firefox.com',
   'testpilot.stage.mozaws.net',
   'testpilot.dev.mozaws.net'
 ];
 
+let RESTART_NEEDED;
+
 function mozAddonManagerInstall(url) {
   return navigator.mozAddonManager.createInstall({ url }).then(install => {
+    navigator.mozAddonManager.addEventListener('onInstalling', data => {
```

Comment:
Does this need to be paired with a `removeEventListener` once we've finished processing here? Guessing the two subscriptions below get garbage collected after the install in question is over, but would this listener catch *all* install progress events going forward? Probably harmless, but concerned about "dead" code getting called.